contracts/core/test/utils/XSubGen.sol

```diff
@@ -59,18 +62,20 @@ contract XSubGen is Test {
             msgFlags[i] = true;
         }
 
-        return makeXSub(1, xheader, msgs, msgFlags);
+        return makeXSub(valSetId, destChainId, xheader, msgs, msgFlags);
     }
 
     /// @dev Make an xsubmission signed by `valSetId`, with the given xheader and selected xmsgs.
     function makeXSub(
         uint64 valSetId,
+        uint64 destChainId,
         XTypes.BlockHeader memory xheader,
         XTypes.Msg[] memory msgs,
         bool[] memory msgFlags
     ) public view returns (XTypes.Submission memory) {
         require(msgs.length == msgFlags.length, "msg length must match msgFlags length");
 
+        (msgs, msgFlags) = _filterXMsgs(msgs, msgFlags, destChainId);
```

Comment:
Now we're introducing two ways of selecting a subset of `msgs` - `msgFlags`and `destChainId`.

I'd prefer to keep surface are of this function sig simpler. If a caller wants filter by destChain, he can use msgFlags. But he may not always want to do that (ex, testing that a submission with mutiple dest chains revert).

Also, by filtering the `msgs` (rather than just msgFlags) we are creating a submission with an attestation root built only from xmsgs with a that dest. The attestation root will change when selecting new dest chains - this is not reflective of how attestation roots are built in the wild. 

I suggest refactoring so something like this

```solidity
makeXSub(valSetId, xheader, msgs, msgFlagsForDest(msgs, destChainId))
```

Reply:
I've refactored to use this flow instead, and adjusted all relevant tests to use it where needed.